### DISCUSSION TITLE
how to use a custom trained model (hugging face) ?

### DISCUSSION THREAD

**Discussion by sarthusarth** [jalammar/ecco#72] [Q&A] [answered]
I was not able to find documentation for using the library for using custom models loading from the folder doesn't work. Where can I find it?

**Comment by sarthusarth** [comment, marked as answer]
sorry for asking found it, incase it helps anyone here is the link https://ecco.readthedocs.io/en/main/api/ecco/

**Reply by AmirBialer** [reply]
I looked at the link above and didn't find an answer:

I have a model which is based on Hugging Face's BERT, wrapped by a custom class which was finetuned for text classification.
Is there a way to load this custom class into ecco?
thanks

**Reply by nicedlp** [reply]
@AmirBialer 
I suppose you have found a solution by now, but in case someone else needs.

Imagine you downloaded BERT from HuggingFace and finetuned it on your own corpus. To use ecco, you replace the model name with the path of the folder containing the model on your computer.
`python 
lm = ecco.from_pretrained(model_folder_path, activations=True, model_config=model_config)
` 
If you need help with creating the model_config dictionary, ecco authors created a [notebook just for that](https://github.com/jalammar/ecco/blob/main/notebooks/Identifying%20model%20configuration.ipynb)

**Reply by AmirBialer** [reply]
thanks for your input @nicedlp 

By saying I use a custom architecture I mean something like this (for some reason I can't write the entire code in code format)

'class BERT_Arch(nn.Module):

       def __init__(self, bert, num_labels=2):
          super(BERT_Arch, self).__init__()
          self.bert = bert 
          # dropout layer
          self.dropout = nn.Dropout(0.1)
          # dense layer 1
          self.DenseClassifier1 = nn.Linear(768,num_labels)      
          #softmax activation function
          self.softmax = nn.LogSoftmax(dim=1)            
   
        def forward(self, sent_id, mask):         #define the forward pass
          #pass the inputs to the model  
          cls_hs=self.bert(sent_id,mask)[0][:,0,:]    #coreponds to cls token 
          x=self.DenseClassifier1(cls_hs)
          x = self.dropout(x)      
          x = self.softmax(x)
          return x
'

is there any way to use my BERT_Arch with ecco?
or must the architecture be one defined by hugging face?
thanks!
